src/scripts/components/DateRangeDropdown.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Dropdown, DropdownMenuItem, DropdownMenuDivider, DropdownMenuItemWild, Date, Button, ResourceRight } from '.';
+
+class DateRangeDropdown extends React.Component {
+  state = {
+    isApplyEnabled: false,
+    startDate: moment(this.props.startDate),
+    endDate: moment(this.props.endDate),
+    minDate: moment(this.props.minDate),
+    maxDate: moment(this.props.maxDate),
+    dropdownMenuItems: this.props.dropdownMenuItems || [],
+    activeKey: this.props.activeKey,
+  };
+
+  componentDidMount() {
+    this.setState({ isApplyEnabled: false, startDate: moment(this.props.startDate), endDate: moment(this.props.endDate) });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.endDate !== this.state.endDate && this.props.isCustomDate) {
+      this.endDatePickerInput.datePickerInput.input.setRawValue(this.state.endDate.format('MM/DD/YYYY'));
+    }
+    if (prevState.startDate !== this.state.startDate && this.props.isCustomDate) {
+      this.startDatePickerInput.datePickerInput.input.setRawValue(this.state.startDate.format('MM/DD/YYYY'));
+    }
+  }
+
+  handleDropdownClick = () => {
+    if (this.props.isCustomDate) {
+      const startInputValue = this.startDatePickerInput.datePickerInput.input.element.value;
+      const endInputValue = this.endDatePickerInput.datePickerInput.input.element.value;
+      const { startDate, endDate } = this.props;
+      if (this.state.isApplyEnabled || endInputValue !== moment(endDate).format('MM/DD/YYYY') || startInputValue !== moment(startDate).format('MM/DD/YYYY')) {
+        this.endDatePickerInput.datePickerInput.input.element.value = moment(this.props.endDate).format('MM/DD/YYYY');
```

Comment:
can you make this line a descriptive constant declaration for readability?

`this.state.isApplyEnabled || endInputValue !== moment(endDate).format('MM/DD/YYYY') || startInputValue !== moment(startDate).format('MM/DD/YYYY')`

Reply:
My thought is same as above for this as well.

Reply:
Still would prefer descriptive constants, but would be fine with a comment in the file to help identify what's going on